electron/libelectron/src/backends/photon.cc

```diff
@@ -11,14 +11,19 @@ namespace electron {
 
 PhotonBackend::PhotonBackend() :
     lib_handle{nullptr},
-    emu(std::make_shared<photon::PhotonEmu>()) { }
+    emu(std::make_shared<photon::PhotonEmu>()) {
+        device_memory = mmap(NULL, (unsigned long int) 8 * (1 << 30), PROT_READ | PROT_WRITE, MAP_ANONYMOUS | MAP_SHARED, 0, 0);
+        assert(device_memory != MAP_FAILED);
+        allocator=std::make_shared<memory_mgmt::FirstFitAllocator>((unsigned long int) 8 * (1 << 30), device_memory);
+    }
```

Comment:
Please use c++ initializer list for this:
```cpp
PhotonBackend::PhotonBackend() :
    lib_handle{nullptr},
    emu(std::make_shared<photon::PhotonEmu>()),
    allocator(std::make_shared<memory_mgmt::FirstFitAllocator>((unsigned long int) 8 * (1 << 30))),
    device_memory(mmap(NULL, (unsigned long int) 8 * (1 << 30), PROT_READ | PROT_WRITE, MAP_ANONYMOUS | MAP_SHARED, -1, 0))
{
    assert(device_memory != MAP_FAILED);
}
```

Also, set the mmap fd parameter to -1 (See my change)

Reply:
Done